osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/AttributeAddressHelperService.java

```diff
@@ -0,0 +1,248 @@
+            throws ProtocolAdapterException {
+
+        final SelectiveAccessDescription access = this.getSelectiveAccessDescription(periodType, beginDateTime,
+                endDateTime, isSelectingValuesSupported);
+
+        final List<AttributeAddress> profileBuffer = new ArrayList<>();
+        switch (periodType) {
+        case INTERVAL:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_INTERVAL_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        case DAILY:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_DAILY_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        case MONTHLY:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_MONTHLY_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        default:
+            throw new ProtocolAdapterException(String.format("periodtype %s not supported", periodType));
+        }
+        profileBuffer.addAll(this.getScalerUnit(periodType));
+        return profileBuffer.toArray(new AttributeAddress[profileBuffer.size()]);
+    }
+
+    private List<AttributeAddress> getScalerUnit(final PeriodTypeDto periodType) throws ProtocolAdapterException {
+        switch (periodType) {
+        case INTERVAL:
+            return this.createScalerUnitForInterval();
+        case DAILY:
+        case MONTHLY:
+            return this.createScalerUnitForMonth();
+        default:
+            throw new ProtocolAdapterException(String.format("periodtype %s not supported", periodType));
+        }
+    }
+
+    private List<AttributeAddress> createScalerUnitForInterval() {
+        final List<AttributeAddress> scalerUnit = new ArrayList<>();
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_INTERVAL_IMPORT_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_INTERVAL_EXPORT_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
+        return scalerUnit;
+    }
+
+    private List<AttributeAddress> createScalerUnitForMonth() {
+        final List<AttributeAddress> scalerUnit = new ArrayList<>();
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_1_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_2_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_1_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
+        scalerUnit.add(new AttributeAddress(CLASS_ID_REGISTER, OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_2_SCALER_UNIT,
+                ATTRIBUTE_ID_SCALER_UNIT));
```

Comment:
The lists of AttributeAddress created in this class seem to be constants in the sense that they have only final fields and are instantiated from fixed values. I suggest moving them out into a separate class in which they are declared once as public fields. That makes their role more clear and avoid having to instantiate them over and over.

Reply:
Moved to the constants class